### PR TITLE
Support Task Operations and shell commands for task logs

### LINKED ISSUE #3390: TaskOperations needs to support the retrieval of logs
A user can pass in a task execution id and retrieve a string representation of a log from taskOperations.
Also needs to support a shell call such as `task execution log --id`.

This is required to support https://github.com/spring-cloud/spring-cloud-dataflow-acceptance-tests/issues/210

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #3390

This is to support Acceptance Tests for validating how CTR handles properties

### CODE REVIEW

**Comment by cppwfs** [comment]
Closing PR per request.

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskOperations.java [hunk 1]
```diff
@@ -105,6 +105,23 @@
 	 */
 	TaskExecutionResource taskExecutionStatus(long id);
 
+	/**
+	 * Return the task execution log.  The platform to from which to retrieve the log will be set to {@code default}.
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/TaskOperations.java]: we can remove `to`

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java [hunk 2]
```diff
@@ -103,6 +107,65 @@ public long launchWithAlternateCTR(String taskName, String ctrAppName) {
...
+		CommandResult idResult = shell.executeCommand("task execution list --name " + taskName);
+		Table taskExecutionResult = (Table) idResult.getResult();
+
+		long id = (long) taskExecutionResult.getModel().getValue(1, 1);
+		assertTrue(cr.toString().contains("with execution id " + id));
+		waitForDBToBePopulated(id);
+		return id;
+	}
+
+	private void waitForDBToBePopulated(long id) throws Exception {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java]: Not sure why do we need this method.

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java [hunk 3]
```diff
@@ -103,6 +107,65 @@ public long launchWithAlternateCTR(String taskName, String ctrAppName) {
...
 
+	/**
+	 * Launch a task and validate the result from shell on default platform.
+	 *
+	 * @param taskName the name of the task
+	 */
+	public String getTaskExecutionLog(String taskName) throws Exception{
+		long id = launchTaskExecutionForLog(taskName);
+		CommandResult cr = shell.executeCommand("task execution log --id " + id);
+		assertTrue(cr.toString().contains("Starting TimestampTaskApplication"));
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java]: Checking `TimestampTaskApplication` here may not be correct as this method looks generic.

File: spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java (hunk 2 above)
Comment (ilayaperumalg) [reply spring-cloud-dataflow-shell-core/src/test/java/org/springframework/cloud/dataflow/shell/command/TaskCommandTemplate.java]: sorry, I get this now.

File: spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/TaskCommands.java [hunk 4]
```diff
@@ -229,6 +230,20 @@ public String stop(@CliOption(key = { "", "ids" }, help = "the task execution id
 		return String.format("Request to stop the task execution with id(s): %s has been submitted", ids);
 	}
 
+	@CliCommand(value = LOG, help = "Retrieve task execution log")
+	public String retrieveTaskExecutionLog(@CliOption(key = { "", "id" }, help = "the task execution id", mandatory = true) long id,
+			@CliOption(key = { "", "platform" }, help = "the platform of the task execution", mandatory = false) String platform) {
```
Comment (ilayaperumalg) [inline spring-cloud-dataflow-shell-core/src/main/java/org/springframework/cloud/dataflow/shell/command/TaskCommands.java]: We can remove key `""` option for the platform key as we can only have one of the options with the default empty key.

**Comment by ilayaperumalg** [comment]
LGTM, fixing the above minor comments and merging the PR.

**Comment by ilayaperumalg** [comment]
Rebased and merged as d324ebed452765c661287dde8a779595dfb1c416

**Pull request closed without merging**